developers/products/interchain-messaging-agent/enabling-tokens-on-schains.adoc

```diff
@@ -0,0 +1,126 @@
+= IMA Token Access Control
+:icons: font
+:toc: macro
+
+ifdef::env-github[]
+
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+
+endif::[]
+
+toc::[]
+
+== Introduction
+
+The SKALE Chain owner has special permissions to setup specific IMA parameters, such as IMA access control. These permissions include:
+
+* Enable/disable whitelist on mainnet (default: enabled)
+* Enable/disable automatic deployment on schain (default: disabled)
+* AddERC20TokenByOwner on mainnet and on schain
+* AddERC721TokenByOwner on mainnet and on schain
+
+== Enable or Disable the Whitelist
+
+The token whitelist allows the Schain owner to allow token exchange through IMA and a SKALE chain. For example, a decentralized exchange may prefer to disable the whitelist to allow any token transfer. A dApp using only one token may enable the whitelist to allow only a single token exchange. 
+
+To disable the whitelist, the owner executes `disableWhitelist` in LockAndDataForMainnetERC20 or LockAndDataForMainnetERC721 contracts, and pass in the schainName as the only argument. For example:
+
+```python
+lock_and_data_for_mainnet_erc20 = self._get_contract_on_mainnet('lock_and_data_for_mainnet_erc20')
+disable = lock_and_data_for_mainnet_erc20.encodeABI(fn_name="disableWhitelist", args=[schainName])
+
+signed_txn = self.web3_mainnet.eth.account.signTransaction(dict(
+        nonce=self.web3_mainnet.eth.getTransactionCount(sender_address),
+        gasPrice=self.web3_mainnet.eth.gasPrice,
+        gas=200000,
+        to=lock_and_data_for_mainnet_erc20.address,
+        value=0,
+        data = disable
+    ),
+    from_key)
+
+self.web3_mainnet.eth.sendRawTransaction(signed_txn.rawTransaction)
+```
+
+To re-enable the whitelist, the owner executes `enableWhitelist` as demonstrated above.
```

Comment:
fn_name="enableWhitelist"

Reply:
and what about the `data=disable` during enable?

Reply:
I'll add codeblock for clarity